core/src/test/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseTest.java

```diff
@@ -341,4 +344,36 @@ void constructionHeuristicAllocateToValueFromQueue() {
         assertThat(solution.getScore().isSolutionInitialized()).isTrue();
     }
 
+    @Test
+    void constructionHeuristicWithDifficultyWeight() {
+        // Default configuration
+        var solverConfig = PlannerTestUtils.buildSolverConfig(TestdataDifficultyWeightSolution.class,
+                TestdataDifficultyWeightEntity.class);
+        solverConfig.setPhaseConfigList(null);
+        var solution = new TestdataDifficultyWeightSolution("s1");
+        solution.setValueList(
+                List.of(new TestdataDifficultyWeightValue("v1"),
+                        new TestdataDifficultyWeightValue("v2")));
+        solution.setEntityList(List.of(new TestdataDifficultyWeightEntity("e1")));
+        solution = PlannerTestUtils.solve(solverConfig, solution);
+        assertThat(solution).isNotNull();
+        assertThat(solution.getEntityList().stream().allMatch(TestdataDifficultyWeightEntity::isComparisonCalled))
+                .isTrue();
+
+        // Custom phase configuration
+        solverConfig = PlannerTestUtils.buildSolverConfig(TestdataDifficultyWeightSolution.class,
+                TestdataDifficultyWeightEntity.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig());
+
+        solution = new TestdataDifficultyWeightSolution("s1");
+        solution.setValueList(
+                List.of(new TestdataDifficultyWeightValue("v1"),
+                        new TestdataDifficultyWeightValue("v2")));
+        solution.setEntityList(List.of(new TestdataDifficultyWeightEntity("e1")));
+        solution = PlannerTestUtils.solve(solverConfig, solution);
```

Comment:
I'll be honest - I don't like this. We can't be testing everything by running the solver. It takes too much time, and the tests get bloated.

I'd verify that this was detected on the Entity Descriptor, and that's that.